src/test/java/seedu/address/model/concert/UniqueConcertContactListTest.java

```diff
@@ -0,0 +1,191 @@
+package seedu.address.model.concert;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.testutil.TypicalConcertContacts.ALICE_COACHELLA;
+import static seedu.address.testutil.TypicalConcertContacts.BOB_COACHELLA;
+import static seedu.address.testutil.TypicalConcerts.COACHELLA;
+import static seedu.address.testutil.TypicalPersons.ALICE;
+import static seedu.address.testutil.TypicalPersons.BOB;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.model.concert.exceptions.ConcertContactNotFoundException;
+import seedu.address.model.concert.exceptions.DuplicateConcertContactException;
+import seedu.address.model.person.Person;
+
+public class UniqueConcertContactListTest {
+    private final UniqueConcertContactList uniqueConcertContactList =
+            new UniqueConcertContactList();
+
+    @Test
+    public void contains_nullConcertContact_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> uniqueConcertContactList.contains(null));
+    }
+
+    @Test
+    public void contains_concertContactNotInList_returnsFalse() {
+        assertFalse(uniqueConcertContactList.contains(ALICE_COACHELLA));
+    }
+
+    @Test
+    public void contains_person_concert() {
+        uniqueConcertContactList.add(BOB_COACHELLA);
+
+        // null throws NullPointerException
+        assertThrows(NullPointerException.class, () -> uniqueConcertContactList.contains(
+                (Person) null, (Concert) null));
+
+        // inside returns true
+        assertTrue(uniqueConcertContactList.contains(BOB, COACHELLA));
+
+        // not inside returns false
+        assertFalse(uniqueConcertContactList.contains(ALICE, COACHELLA));
+    }
+
+    @Test
+    public void contains_concertContactInList_returnsTrue() {
+        uniqueConcertContactList.add(ALICE_COACHELLA);
+        assertTrue(uniqueConcertContactList.contains(ALICE_COACHELLA));
+    }
+
+    @Test
+    public void add_nullConcertContact_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> uniqueConcertContactList.add(null));
+    }
+
+    @Test
+    public void add_duplicateConcertContact_throwsDuplicateConcertException() {
+        uniqueConcertContactList.add(ALICE_COACHELLA);
+        assertThrows(DuplicateConcertContactException.class, () -> uniqueConcertContactList.add(
+                ALICE_COACHELLA));
+    }
+
```

Comment:
missing success test for add method. Can use this
@Test
    public void add_success() {
        uniqueConcertContactList.add(ALICE_COACHELLA);

        // change email field
        Person editedAlice = new PersonBuilder(ALICE).withEmail("new_email@example.com").build();
        ConcertContact cc = new ConcertContactBuilder(ALICE_COACHELLA).withPerson(editedAlice).build();
        assertDoesNotThrow(() -> uniqueConcertContactList.add(cc));
    }